File: HW/HW 1/Solution.java
public class Solution{
    public boolean isPalindrome(String s){
        s = s.toLowerCase().replaceAll("[^a-z0-9]", "");
        int left = 0;
        int right = s.length()-1;
        while(left<right){
            if(s.charAt(left) != s.charAt(right)) return false;
            left++;
            right--;
        }
        return true;
    }
    public static void main(String[] args){
        Solution solution = new Solution();
        String s1 = "A man, nama";
        String s2 = "runner is a rennur";
        System.out.println(solution.isPalindrome(s1));
        System.out.println(solution.isPalindrome(s2));
    }
}

File: HW/HW 1/Solution.py
class Solution:
    def isPalindrome(self, s: str)->bool:
        s = ''.join(char.lower() for char in s if char.isalnum())
        left, right = 0, len(s)-1
        while left<right:
            if s[left] != s[right]: return False
            right -=1
            left +=1
        return True
    
solution = Solution()
s1 = "A man, nama"
s2 = "runner is a rennur"
print(solution.isPalindrome(s1))
print(solution.isPalindrome(s2))